app/assets/javascripts/student_profile_v2/chart.js

```diff
@@ -0,0 +1,77 @@
+(function() {
+  window.shared || (window.shared = {});
+  var dom = window.shared.ReactHelpers.dom;
+  var createEl = window.shared.ReactHelpers.createEl;
+  var merge = window.shared.ReactHelpers.merge;
+
+  var ProfileChartSettings = window.ProfileChartSettings;
+  var HighchartsWrapper = window.shared.HighchartsWrapper;
+
+  // Component for all charts in the profile page.
+  window.shared.Chart = React.createClass({
```

Comment:
Nice description and comments here too!  The `Chart` name is a bit generic, I'd recommend something a bit more explicit, maybe something like `ProfileHistoricalChart`?